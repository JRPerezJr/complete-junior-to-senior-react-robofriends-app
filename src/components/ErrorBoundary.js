import React, { Component } from 'react';

export default class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
    };
  }
  componentDidCatch(error, info) {
    this.setState({ hasError: true });
  }
  render() {
    const { hasError } = this.state;
    return hasError ? (
      <h1>Oh snap! Something went bad.</h1>
    ) : (
      this.props.children
    );
  }
}
