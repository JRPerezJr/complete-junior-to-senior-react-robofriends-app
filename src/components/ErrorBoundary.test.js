import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('<ErrorBoundary/> component tests', () => {
  const wrapper = shallow(<ErrorBoundary />);
  it('sanity test', () => {
    expect(true).toBe(true);
  });

  it('renders <ErrorBoundary/> component', () => {
    expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
  });

  it('checks ErrorBoundary state before error', () => {
    expect(wrapper.state()).toEqual({ hasError: false });
  });

  it('displays error message on error generated by child', () => {
    const Something = () => null;

    const errorWrapper = shallow(
      <ErrorBoundary>
        <Something />
      </ErrorBoundary>
    );
    const error = new Error('test');
    errorWrapper.find(Something).simulateError(error);

    expect(errorWrapper.state()).toEqual({ hasError: true });
    expect(errorWrapper).toMatchSnapshot();
  });
});
