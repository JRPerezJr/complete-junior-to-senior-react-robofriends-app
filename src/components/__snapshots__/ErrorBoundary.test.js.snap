// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary/> component tests displays error message on error generated by child 1`] = `ShallowWrapper {}`;

exports[`<ErrorBoundary/> component tests renders <ErrorBoundary/> component 1`] = `ShallowWrapper {}`;
